plugins/vendor/zoom/ZoomDropFolderPlugin/batch/KZoomDropFolderEngine.php

```diff
@@ -0,0 +1,423 @@
+			$entryPager->pageIndex = 1;
+			KBatchBase::impersonate($this->dropFolder->partnerId);
+			$entryList = KBatchBase::$kClient->baseEntry->listAction($entryFilter, $entryPager);
+			KBatchBase::unimpersonate();
+			if (is_array($entryList->objects) && isset($entryList->objects[0]) )
+			{
+				return $entryList->objects[0];
+			}
+			
+			return null;
+		}
+		catch (Exception $e)
+		{
+			KalturaLog::err('Failed to get entry by reference id: '. $referenceId . $e->getMessage() );
+			return null;
+		}
+	}
+	
+	protected function createEntry($uuid)
+	{
+		$newEntry = new KalturaMediaEntry();
+		$newEntry->sourceType = KalturaSourceType::URL;
+		$newEntry->mediaType = KalturaMediaType::VIDEO;
+		$newEntry->referenceId = zoomProcessor::ZOOM_PREFIX . $uuid;
+		KBatchBase::impersonate($this->dropFolder->partnerId);
+		$entry = KBatchBase::$kClient->baseEntry->add($newEntry);
+		KBatchBase::unimpersonate();
+		return $entry;
+	}
+	
+	protected static function orderRecordingFiles($recordingFiles)
+	{
+		$recordingFilesOrdered = array();
+		foreach($recordingFiles as $recordingFile)
+		{
+			if(!isset($recordingFilesOrdered[$recordingFile[self::RECORDING_START]]))
+			{
+				$recordingFilesOrdered[$recordingFile[self::RECORDING_START]] = array();
+			}
+			$recordingFilesOrdered[$recordingFile[self::RECORDING_START]][] = $recordingFile;
+		}
+		ksort($recordingFilesOrdered);
+		return self::orderRecordingFilesByType($recordingFilesOrdered);
+	}
+	
+	protected static function orderRecordingFilesByType($recordingFilesOrdered)
+	{
+		foreach ($recordingFilesOrdered as $time => $recordingFilesPerTimeSlot)
+		{
+			$filesOrderByType = array();
+			foreach ($recordingFilesPerTimeSlot as $recordingFile)
+			{
+				if(!isset($filesOrderByType[$recordingFile[self::RECORDING_FILE_TYPE]]))
+				{
+					$filesOrderByType[$recordingFile[self::RECORDING_FILE_TYPE]] = array();
+				}
+				$filesOrderByType[$recordingFile[self::RECORDING_FILE_TYPE]][] = $recordingFile;
+			}
+			ksort($filesOrderByType);
+			$byTime = array();
+			foreach ($filesOrderByType as $fileOrderByType)
+			{
+				foreach ($fileOrderByType as $file)
+				{
+					$byTime[] = $file; //flatten the array - removing the key type
+				}
+			}
+			$recordingFilesOrdered[$time] = $byTime;
+		}
+		return $recordingFilesOrdered;
+	}
+
+	protected function updateDropFolderLastMeetingHandled($lastHandledMeetingTime)
+	{
+		$updateDropFolder = new KalturaZoomDropFolder();
+		$updateDropFolder->lastHandledMeetingTime = $lastHandledMeetingTime;
+		try
+		{
+			$this->dropFolderPlugin->dropFolder->update($this->dropFolder->id, $updateDropFolder);
+			KalturaLog::debug('Last handled meetings time is: '. $lastHandledMeetingTime);
+		}
+		catch(Exception $e)
+		{
+			KalturaLog::debug('Cannot update drop folder with last Meeting Handled Time - '.$e->getMessage());
+			return null;
+		}
+	}
+
+	protected function addDropFolderFile($meetingFile, $recordingFile, $entryId, $isParentEntry = false)

```

Comment:
arguments name are confusing,
 $entryId is actual the parent entry Id - please rename.